third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -228,6 +228,7 @@ ComputationClient::DataPtr PjRtComputationClient::TransferShardsToServer(
     xla::Shape shape, xla::OpSharding sharding) {
   TF_VLOG(1) << "TransferShardsToServer with " << tensor_shards.size()
              << " shards.";
+  // TODO(jonbolin): Consider using CopyToDevice when sharding is REPLICATED
```

Comment:
This happens even after the InputHandler, rather, don't we want to move this to the earlier steps?

Reply:
Let's also explain why we are opting out of copytodevice for now, but also note that copytodevice without blocking is faster than async data transfer from the host.